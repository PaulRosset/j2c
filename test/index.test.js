import J2C from "../lib/main";
import Papa from "papaparse";
const fs = require("fs");

let jsonAPI;
let jsonParse;
let csvPretty;

describe("Testing Api", () => {
  beforeAll(() => {
    const jsonFromFile = fs.readFileSync(`${__dirname}/test.json`);
    jsonAPI = new J2C();
    jsonParse = JSON.parse(jsonFromFile);
  });

  test("Test normal functionality", () => {
    const csv = jsonAPI.convert(jsonParse);
    csvPretty = jsonAPI.onceFilled();
    expect(csvPretty).toMatchSnapshot();
  });

  test("Testing linter csv", done => {
    Papa.parse("file.csv", {
      complete: results => {
        expect(results).toMatchSnapshot();
        done();
      }
    });
  });

  describe("Further tests", () => {
    test("Length of both JSON-CSV should be equal", () => {
      const sizeJson = jsonParse.length;
      expect(sizeJson).toEqual(100);
      const sizeCsv = csvPretty.split(".").filter((item, pos, self) => {
        return !isNaN(item) && self.indexOf(item) === pos;
      });
      expect(sizeCsv.length).toEqual(sizeJson);
    });

    test("Verify length of both", () => {
      const sizeObject = jsonParse
        .map(polo => Object.keys(polo).length)
        .filter((item, pos, self) => self.indexOf(item) === pos);
      expect(sizeObject.length).toEqual(1);
      expect(sizeObject[0]).toEqual(4);
      const sizeCsvObject = csvPretty
        .split(".")
        .filter(item => !isNaN(item))
        .reduce((acc, currentValue) => {
          if (typeof acc[currentValue] == "undefined") {
            acc[currentValue] = 1;
          } else {
            acc[currentValue] += 1;
          }
          return acc;
        }, [])
        .filter((item, pos, self) => self.indexOf(item) === pos);
      expect(sizeCsvObject.length).toEqual(sizeObject.length);
      expect(sizeCsvObject[0]).toEqual(sizeObject[0]);
    });
  });
});
