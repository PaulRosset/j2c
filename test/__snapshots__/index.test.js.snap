// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Api Test normal functionality 1`] = `
"Key,Values
obj.0.userId,1
obj.0.id,1
obj.0.title,sunt aut facere repellat provident occaecati excepturi optio reprehenderit
obj.0.body,quia et suscipitsuscipit recusandae consequuntur expedita et cumreprehenderit molestiae ut ut quas totamnostrum rerum est autem sunt rem eveniet architecto
obj.1.userId,1
obj.1.id,2
obj.1.title,qui est esse
obj.1.body,est rerum tempore vitaesequi sint nihil reprehenderit dolor beatae ea dolores nequefugiat blanditiis voluptate porro vel nihil molestiae ut reiciendisqui aperiam non debitis possimus qui neque nisi nulla
obj.2.userId,1
obj.2.id,3
obj.2.title,ea molestias quasi exercitationem repellat qui ipsa sit aut
obj.2.body,et iusto sed quo iurevoluptatem occaecati omnis eligendi aut advoluptatem doloribus vel accusantium quis pariaturmolestiae porro eius odio et labore et velit aut
obj.3.userId,1
obj.3.id,4
obj.3.title,eum et est occaecati
obj.3.body,ullam et saepe reiciendis voluptatem adipiscisit amet autem assumenda provident rerum culpaquis hic commodi nesciunt rem tenetur doloremque ipsam iurequis sunt voluptatem rerum illo velit
obj.4.userId,1
obj.4.id,5
obj.4.title,nesciunt quas odio
obj.4.body,repudiandae veniam quaerat sunt sedalias aut fugiat sit autem sed estvoluptatem omnis possimus esse voluptatibus quisest aut tenetur dolor neque
obj.5.userId,1
obj.5.id,6
obj.5.title,dolorem eum magni eos aperiam quia
obj.5.body,ut aspernatur corporis harum nihil quis provident sequimollitia nobis aliquid molestiaeperspiciatis et ea nemo ab reprehenderit accusantium quasvoluptate dolores velit et doloremque molestiae
obj.6.userId,1
obj.6.id,7
obj.6.title,magnam facilis autem
obj.6.body,dolore placeat quibusdam ea quo vitaemagni quis enim qui quis quo nemo aut saepequidem repellat excepturi ut quiasunt ut sequi eos ea sed quas
obj.7.userId,1
obj.7.id,8
obj.7.title,dolorem dolore est ipsam
obj.7.body,dignissimos aperiam dolorem qui eumfacilis quibusdam animi sint suscipit qui sint possimus cumquaerat magni maiores excepturiipsam ut commodi dolor voluptatum modi aut vitae
obj.8.userId,1
obj.8.id,9
obj.8.title,nesciunt iure omnis dolorem tempora et accusantium
obj.8.body,consectetur animi nesciunt iure doloreenim quia adveniam autem ut quam aut nobiset est aut quod aut provident voluptas autem voluptas
obj.9.userId,1
obj.9.id,10
obj.9.title,optio molestias id quia eum
obj.9.body,quo et expedita modi cum officia vel magnidoloribus qui repudiandaevero nisi sitquos veniam quod sed accusamus veritatis error
obj.10.userId,2
obj.10.id,11
obj.10.title,et ea vero quia laudantium autem
obj.10.body,delectus reiciendis molestiae occaecati non minima eveniet qui voluptatibusaccusamus in eum beatae sitvel qui neque voluptates ut commodi qui inciduntut animi commodi
obj.11.userId,2
obj.11.id,12
obj.11.title,in quibusdam tempore odit est dolorem
obj.11.body,itaque id aut magnampraesentium quia et ea odit et ea voluptas etsapiente quia nihil amet occaecati quia id voluptatemincidunt ea est distinctio odio
obj.12.userId,2
obj.12.id,13
obj.12.title,dolorum ut in voluptas mollitia et saepe quo animi
obj.12.body,aut dicta possimus sint mollitia voluptas commodi quo doloremqueiste corrupti reiciendis voluptatem eius rerumsit cumque quod eligendi laborum minimaperferendis recusandae assumenda consectetur porro architecto ipsum ipsam
obj.13.userId,2
obj.13.id,14
obj.13.title,voluptatem eligendi optio
obj.13.body,fuga et accusamus dolorum perferendis illo voluptasnon doloremque neque faceread qui dolorum molestiae beataesed aut voluptas totam sit illum
obj.14.userId,2
obj.14.id,15
obj.14.title,eveniet quod temporibus
obj.14.body,reprehenderit quos placeatvelit minima officia dolores impedit repudiandae molestiae namvoluptas recusandae quis delectusofficiis harum fugiat vitae
obj.15.userId,2
obj.15.id,16
obj.15.title,sint suscipit perspiciatis velit dolorum rerum ipsa laboriosam odio
obj.15.body,suscipit nam nisi quo aperiam autasperiores eos fugit maiores voluptatibus quiavoluptatem quis ullam qui in alias quia estconsequatur magni mollitia accusamus ea nisi voluptate dicta
obj.16.userId,2
obj.16.id,17
obj.16.title,fugit voluptas sed molestias voluptatem provident
obj.16.body,eos voluptas et aut odit natus earumaspernatur fuga molestiae ullamdeserunt ratione qui eosqui nihil ratione nemo velit ut aut id quo
obj.17.userId,2
obj.17.id,18
obj.17.title,voluptate et itaque vero tempora molestiae
obj.17.body,eveniet quo quislaborum totam consequatur non dolorut et est repudiandaeest voluptatem vel debitis et magnam
obj.18.userId,2
obj.18.id,19
obj.18.title,adipisci placeat illum aut reiciendis qui
obj.18.body,illum quis cupiditate provident sit magnamea sed aut omnisveniam maiores ullam consequatur atqueadipisci quo iste expedita sit quos voluptas
obj.19.userId,2
obj.19.id,20
obj.19.title,doloribus ad provident suscipit at
obj.19.body,qui consequuntur ducimus possimus quisquam amet similiquesuscipit porro ipsam ameteos veritatis officiis exercitationem vel fugit aut necessitatibus totamomnis rerum consequatur expedita quidem cumque explicabo
obj.20.userId,3
obj.20.id,21
obj.20.title,asperiores ea ipsam voluptatibus modi minima quia sint
obj.20.body,repellat aliquid praesentium dolorem quosed totam minus non itaquenihil labore molestiae sunt dolor eveniet hic recusandae veniamtempora et tenetur expedita sunt
obj.21.userId,3
obj.21.id,22
obj.21.title,dolor sint quo a velit explicabo quia nam
obj.21.body,eos qui et ipsum ipsam suscipit autsed omnis non odioexpedita earum mollitia molestiae aut atque rem suscipitnam impedit esse
obj.22.userId,3
obj.22.id,23
obj.22.title,maxime id vitae nihil numquam
obj.22.body,veritatis unde neque eligendiquae quod architecto quo neque vitaeest illo sit tempora doloremque fugit quodet et vel beatae sequi ullam sed tenetur perspiciatis
obj.23.userId,3
obj.23.id,24
obj.23.title,autem hic labore sunt dolores incidunt
obj.23.body,enim et ex nullaomnis voluptas quia quivoluptatem consequatur numquam aliquam sunttotam recusandae id dignissimos aut sed asperiores deserunt
obj.24.userId,3
obj.24.id,25
obj.24.title,rem alias distinctio quo quis
obj.24.body,ullam consequatur utomnis quis sit vel consequunturipsa eligendi ipsum molestiae et omnis error nostrummolestiae illo tempore quia et distinctio
obj.25.userId,3
obj.25.id,26
obj.25.title,est et quae odit qui non
obj.25.body,similique esse doloribus nihil accusamusomnis dolorem fuga consequuntur reprehenderit fugit recusandae temporibusperspiciatis cum ut laudantiumomnis aut molestiae vel vero
obj.26.userId,3
obj.26.id,27
obj.26.title,quasi id et eos tenetur aut quo autem
obj.26.body,eum sed dolores ipsam sint possimus debitis occaecatidebitis qui qui etut placeat enim earum aut odit facilisconsequatur suscipit necessitatibus rerum sed inventore temporibus consequatur
obj.27.userId,3
obj.27.id,28
obj.27.title,delectus ullam et corporis nulla voluptas sequi
obj.27.body,non et quaerat ex quae ad maioresmaiores recusandae totam aut blanditiis mollitia quas illout voluptatibus voluptatemsimilique nostrum eum
obj.28.userId,3
obj.28.id,29
obj.28.title,iusto eius quod necessitatibus culpa ea
obj.28.body,odit magnam ut saepe sed non quitempora atque nihilaccusamus illum doloribus illo doloreligendi repudiandae odit magni similique sed cum maiores
obj.29.userId,3
obj.29.id,30
obj.29.title,a quo magni similique perferendis
obj.29.body,alias dolor cumqueimpedit blanditiis non eveniet odio maximeblanditiis amet eius quis tempora quia autem rema provident perspiciatis quia
obj.30.userId,4
obj.30.id,31
obj.30.title,ullam ut quidem id aut vel consequuntur
obj.30.body,debitis eius sed quibusdam non quis consectetur vitaeimpedit ut qui consequatur sed aut inquidem sit nostrum et maiores adipisci atquequaerat voluptatem adipisci repudiandae
obj.31.userId,4
obj.31.id,32
obj.31.title,doloremque illum aliquid sunt
obj.31.body,deserunt eos nobis asperiores et hicest debitis repellat molestiae optionihil ratione ut eos beatae quibusdam distinctio maioresearum voluptates et aut adipisci ea maiores voluptas maxime
obj.32.userId,4
obj.32.id,33
obj.32.title,qui explicabo molestiae dolorem
obj.32.body,rerum ut et numquam laborum odit est sitid qui sint inquasi tenetur tempore aperiam et quaerat qui inrerum officiis sequi cumque quod
obj.33.userId,4
obj.33.id,34
obj.33.title,magnam ut rerum iure
obj.33.body,ea velit perferendis earum ut voluptatem voluptate itaque iustototam pariatur innemo voluptatem voluptatem autem magni tempora minima inest distinctio qui assumenda accusamus dignissimos officia nesciunt nobis
obj.34.userId,4
obj.34.id,35
obj.34.title,id nihil consequatur molestias animi provident
obj.34.body,nisi error delectus possimus ut eligendi vitaeplaceat eos harum cupiditate facilis reprehenderit voluptatem beataemodi ducimus quo illum voluptas eligendiet nobis quia fugit
obj.35.userId,4
obj.35.id,36
obj.35.title,fuga nam accusamus voluptas reiciendis itaque
obj.35.body,ad mollitia et omnis minus architecto oditvoluptas doloremque maxime aut non ipsa qui alias veniamblanditiis culpa aut quia nihil cumque facere et occaecatiqui aspernatur quia eaque ut aperiam inventore
obj.36.userId,4
obj.36.id,37
obj.36.title,provident vel ut sit ratione est
obj.36.body,debitis et eaque non officia sed nesciunt pariatur velvoluptatem iste vero et eanumquam aut expedita ipsum nulla involuptates omnis consequatur aut enim officiis in quam qui
obj.37.userId,4
obj.37.id,38
obj.37.title,explicabo et eos deleniti nostrum ab id repellendus
obj.37.body,animi esse sit aut sit nesciunt assumenda eum voluptasquia voluptatibus provident quia necessitatibus earerum repudiandae quia voluptatem delectus fugit aut id quiaratione optio eos iusto veniam iure
obj.38.userId,4
obj.38.id,39
obj.38.title,eos dolorem iste accusantium est eaque quam
obj.38.body,corporis rerum ducimus vel eum accusantiummaxime aspernatur a porro possimus iste omnisest in deleniti asperiores fuga autvoluptas sapiente vel dolore minus voluptatem incidunt ex
obj.39.userId,4
obj.39.id,40
obj.39.title,enim quo cumque
obj.39.body,ut voluptatum aliquid illo tenetur nemo sequi quo facilisipsum rem optio mollitia quasvoluptatem eum voluptas quiunde omnis voluptatem iure quasi maxime voluptas nam
obj.40.userId,5
obj.40.id,41
obj.40.title,non est facere
obj.40.body,molestias id nostrumexcepturi molestiae dolore omnis repellendus quaerat saepeconsectetur iste quaerat tenetur asperiores accusamus ex utnam quidem est ducimus sunt debitis saepe
obj.41.userId,5
obj.41.id,42
obj.41.title,commodi ullam sint et excepturi error explicabo praesentium voluptas
obj.41.body,odio fugit voluptatum ducimus earum autem est incidunt voluptatemodit reiciendis aliquam sunt sequi nulla doloremnon facere repellendus voluptates quiaratione harum vitae ut
obj.42.userId,5
obj.42.id,43
obj.42.title,eligendi iste nostrum consequuntur adipisci praesentium sit beatae perferendis
obj.42.body,similique fugit estillum et dolorum harum et voluptate eaque quidemexercitationem quos nam commodi possimus cum odio nihil nulladolorum exercitationem magnam ex et a et distinctio debitis
obj.43.userId,5
obj.43.id,44
obj.43.title,optio dolor molestias sit
obj.43.body,temporibus est consectetur doloreet libero debitis vel velit laboriosam quiaipsum quibusdam qui itaque fuga rem autea et iure quam sed maxime ut distinctio quae
obj.44.userId,5
obj.44.id,45
obj.44.title,ut numquam possimus omnis eius suscipit laudantium iure
obj.44.body,est natus reiciendis nihil possimus aut providentex et dolorrepellat pariatur estnobis rerum repellendus dolorem autem
obj.45.userId,5
obj.45.id,46
obj.45.title,aut quo modi neque nostrum ducimus
obj.45.body,voluptatem quisquam istevoluptatibus natus officiis facilis doloremquis quas ipsamvel et voluptatum in aliquid
obj.46.userId,5
obj.46.id,47
obj.46.title,quibusdam cumque rem aut deserunt
obj.46.body,voluptatem assumenda ut qui ut cupiditate aut impedit veniamoccaecati nemo illum voluptatem laudantiummolestiae beatae rerum ea iure soluta nostrumeligendi et voluptate
obj.47.userId,5
obj.47.id,48
obj.47.title,ut voluptatem illum ea doloribus itaque eos
obj.47.body,voluptates quo voluptatem facilis iure occaecativel assumenda rerum officia etillum perspiciatis ab delenitilaudantium repellat ad ut et autem reprehenderit
obj.48.userId,5
obj.48.id,49
obj.48.title,laborum non sunt aut ut assumenda perspiciatis voluptas
obj.48.body,inventore ab sintnatus fugit id nulla sequi architecto nihil quaerateos tenetur in in eum veritatis nonquibusdam officiis aspernatur cumque aut commodi aut
obj.49.userId,5
obj.49.id,50
obj.49.title,repellendus qui recusandae incidunt voluptates tenetur qui omnis exercitationem
obj.49.body,error suscipit maxime adipisci consequuntur recusandaevoluptas eligendi et est et voluptatesquia distinctio ab amet quaerat molestiae et vitaeadipisci impedit sequi nesciunt quis consectetur
obj.50.userId,6
obj.50.id,51
obj.50.title,soluta aliquam aperiam consequatur illo quis voluptas
obj.50.body,sunt dolores aut doloribusdolore doloribus voluptates tempora etdoloremque et quocum asperiores sit consectetur dolorem
obj.51.userId,6
obj.51.id,52
obj.51.title,qui enim et consequuntur quia animi quis voluptate quibusdam
obj.51.body,iusto est quibusdam fuga quas quaerat molestiasa enim ut sit accusamus enimtemporibus iusto accusantium provident architectosoluta esse reprehenderit qui laborum
obj.52.userId,6
obj.52.id,53
obj.52.title,ut quo aut ducimus alias
obj.52.body,minima harum praesentium eum rerum illo dolorequasi exercitationem rerum namporro quis neque quoconsequatur minus dolor quidem veritatis sunt non explicabo similique
obj.53.userId,6
obj.53.id,54
obj.53.title,sit asperiores ipsam eveniet odio non quia
obj.53.body,totam corporis dignissimosvitae dolorem ut occaecati accusamusex velit deseruntet exercitationem vero incidunt corrupti mollitia
obj.54.userId,6
obj.54.id,55
obj.54.title,sit vel voluptatem et non libero
obj.54.body,debitis excepturi ea perferendis harum libero optioeos accusamus cum fuga ut sapiente repudiandaeet ut incidunt omnis molestiaenihil ut eum odit
obj.55.userId,6
obj.55.id,56
obj.55.title,qui et at rerum necessitatibus
obj.55.body,aut est omnis doloresneque rerum quod ea rerum velit pariatur beatae excepturiet provident voluptas corrupticorporis harum reprehenderit dolores eligendi
obj.56.userId,6
obj.56.id,57
obj.56.title,sed ab est est
obj.56.body,at pariatur consequuntur earum quidemquo est laudantium soluta voluptatemqui ullam et estet cum voluptas voluptatum repellat est
obj.57.userId,6
obj.57.id,58
obj.57.title,voluptatum itaque dolores nisi et quasi
obj.57.body,veniam voluptatum quae adipisci idet id quia eos ad et doloremaliquam quo nisi sunt eos impedit errorad similique veniam
obj.58.userId,6
obj.58.id,59
obj.58.title,qui commodi dolor at maiores et quis id accusantium
obj.58.body,perspiciatis et quam ea autem temporibus non voluptatibus quibeatae a earum officia nesciunt dolores suscipit voluptas etanimi doloribus cum rerum quas et magniet hic ut ut commodi expedita sunt
obj.59.userId,6
obj.59.id,60
obj.59.title,consequatur placeat omnis quisquam quia reprehenderit fugit veritatis facere
obj.59.body,asperiores sunt ab assumenda cumque modi velitqui esse omnisvoluptate et fuga perferendis voluptasillo ratione amet aut et omnis
obj.60.userId,7
obj.60.id,61
obj.60.title,voluptatem doloribus consectetur est ut ducimus
obj.60.body,ab nemo optio odiodelectus tenetur corporis similique nobis repellendus rerum omnis facilisvero blanditiis debitis in nesciunt doloribus dicta doloresmagnam minus velit
obj.61.userId,7
obj.61.id,62
obj.61.title,beatae enim quia vel
obj.61.body,enim aspernatur illo distinctio quae praesentiumbeatae alias amet delectus qui voluptate distinctioodit sint accusantium autem omnisquo molestiae omnis ea eveniet optio
obj.62.userId,7
obj.62.id,63
obj.62.title,voluptas blanditiis repellendus animi ducimus error sapiente et suscipit
obj.62.body,enim adipisci aspernatur nemonumquam omnis facere dolorem dolor ex quis temporibus inciduntab delectus culpa quo reprehenderit blanditiis asperioresaccusantium ut quam in voluptatibus voluptas ipsam dicta
obj.63.userId,7
obj.63.id,64
obj.63.title,et fugit quas eum in in aperiam quod
obj.63.body,id velit blanditiiseum ea voluptatemmolestiae sint occaecati est eos perspiciatisincidunt a error provident eaque aut aut qui
obj.64.userId,7
obj.64.id,65
obj.64.title,consequatur id enim sunt et et
obj.64.body,voluptatibus ex essesint explicabo est aliquid cumque adipisci fuga repellat laboremolestiae corrupti ex saepe at asperiores et perferendisnatus id esse incidunt pariatur
obj.65.userId,7
obj.65.id,66
obj.65.title,repudiandae ea animi iusto
obj.65.body,officia veritatis tenetur vero qui itaquesint non rationesed et ut asperiores iusto eos molestiae nostrumveritatis quibusdam et nemo iusto saepe
obj.66.userId,7
obj.66.id,67
obj.66.title,aliquid eos sed fuga est maxime repellendus
obj.66.body,reprehenderit id nostrumvoluptas doloremque pariatur sint et accusantium quia quod aspernaturet fugiat ametnon sapiente et consequatur necessitatibus molestiae
obj.67.userId,7
obj.67.id,68
obj.67.title,odio quis facere architecto reiciendis optio
obj.67.body,magnam molestiae perferendis quisquamqui cum reiciendisquaerat animi amet hic inventoreea quia deleniti quidem saepe porro velit
obj.68.userId,7
obj.68.id,69
obj.68.title,fugiat quod pariatur odit minima
obj.68.body,officiis error culpa consequatur modi asperiores etdolorum assumenda voluptas et vel qui aut vel rerumvoluptatum quisquam perspiciatis quia rerum consequatur totam quassequi commodi repudiandae asperiores et saepe a
obj.69.userId,7
obj.69.id,70
obj.69.title,voluptatem laborum magni
obj.69.body,sunt repellendus quaeest asperiores aut deleniti esse accusamus repellendus quia autquia dolorem undeeum tempora esse dolore
obj.70.userId,8
obj.70.id,71
obj.70.title,et iusto veniam et illum aut fuga
obj.70.body,occaecati a doloribusiste saepe consectetur placeat eum voluptate dolorem etqui quo quia voluptasrerum ut id enim velit est perferendis
obj.71.userId,8
obj.71.id,72
obj.71.title,sint hic doloribus consequatur eos non id
obj.71.body,quam occaecati qui deleniti consecteturconsequatur aut facere quas exercitationem aliquam hic voluptasneque id sunt ut aut accusamussunt consectetur expedita inventore velit
obj.72.userId,8
obj.72.id,73
obj.72.title,consequuntur deleniti eos quia temporibus ab aliquid at
obj.72.body,voluptatem cumque tenetur consequatur expedita ipsum nemo quia explicaboaut eum minima consequaturtempore cumque quae est etet in consequuntur voluptatem voluptates aut
obj.73.userId,8
obj.73.id,74
obj.73.title,enim unde ratione doloribus quas enim ut sit sapiente
obj.73.body,odit qui et et necessitatibus sint veniammollitia amet doloremque molestiae commodi similique magnam et quamblanditiis est itaquequo et tenetur ratione occaecati molestiae tempora
obj.74.userId,8
obj.74.id,75
obj.74.title,dignissimos eum dolor ut enim et delectus in
obj.74.body,commodi non non omnis et voluptas sitautem aut nobis magnam et sapiente voluptatemet laborum repellat qui delectus facilis temporibusrerum amet et nemo voluptate expedita adipisci error dolorem
obj.75.userId,8
obj.75.id,76
obj.75.title,doloremque officiis ad et non perferendis
obj.75.body,ut animi faceretotam iusto temporemolestiae eum aut et dolorem aperiamquaerat recusandae totam odio
obj.76.userId,8
obj.76.id,77
obj.76.title,necessitatibus quasi exercitationem odio
obj.76.body,modi ut in nulla repudiandae dolorum nostrum eosaut consequatur omnisut incidunt est omnis iste et quamvoluptates sapiente aliquam asperiores nobis amet corrupti repudiandae provident
obj.77.userId,8
obj.77.id,78
obj.77.title,quam voluptatibus rerum veritatis
obj.77.body,nobis facilis odit tempore cupiditate quiaassumenda doloribus rerum qui eaillum et qui totamaut veniam repellendus
obj.78.userId,8
obj.78.id,79
obj.78.title,pariatur consequatur quia magnam autem omnis non amet
obj.78.body,libero accusantium et et facere incidunt sit doloremnon excepturi qui quia sed laudantiumquisquam molestiae ducimus estofficiis esse molestiae iste et quos
obj.79.userId,8
obj.79.id,80
obj.79.title,labore in ex et explicabo corporis aut quas
obj.79.body,ex quod dolorem ea eum iure qui provident ametquia qui facere excepturi et repudiandaeasperiores molestias providentminus incidunt vero fugit rerum sint sunt excepturi provident
obj.80.userId,9
obj.80.id,81
obj.80.title,tempora rem veritatis voluptas quo dolores vero
obj.80.body,facere qui nesciunt est voluptatum voluptatem nisisequi eligendi necessitatibus ea at rerum itaqueharum non ratione velit laboriosam quis consequunturex officiis minima doloremque voluptas ut aut
obj.81.userId,9
obj.81.id,82
obj.81.title,laudantium voluptate suscipit sunt enim enim
obj.81.body,ut libero sit aut totam inventore suntporro sint qui sunt molestiaeconsequatur cupiditate qui iste ducimus adipiscidolor enim assumenda soluta laboriosam amet iste delectus hic
obj.82.userId,9
obj.82.id,83
obj.82.title,odit et voluptates doloribus alias odio et
obj.82.body,est molestiae facilis quis tempora numquam nihil quivoluptate sapiente consequatur est quinecessitatibus autem aut ipsa aperiam modi dolore numquamreprehenderit eius rem quibusdam
obj.83.userId,9
obj.83.id,84
obj.83.title,optio ipsam molestias necessitatibus occaecati facilis veritatis dolores aut
obj.83.body,sint molestiae magni a et quoseaque et quasiut rerum debitis similique veniamrecusandae dignissimos dolor incidunt consequatur odio
obj.84.userId,9
obj.84.id,85
obj.84.title,dolore veritatis porro provident adipisci blanditiis et sunt
obj.84.body,similique sed nisi voluptas iusto omnismollitia et quoassumenda suscipit officia magnam sint sed temporaenim provident pariatur praesentium atque animi amet ratione
obj.85.userId,9
obj.85.id,86
obj.85.title,placeat quia et porro iste
obj.85.body,quasi excepturi consequatur iste autem temporibus sed molestiae beataeet quaerat et esse utvoluptatem occaecati et vel explicabo autemasperiores pariatur deserunt optio
obj.86.userId,9
obj.86.id,87
obj.86.title,nostrum quis quasi placeat
obj.86.body,eos et molestiaenesciunt ut adolores perspiciatis repellendus repellat aliquidmagnam sint rem ipsum est
obj.87.userId,9
obj.87.id,88
obj.87.title,sapiente omnis fugit eos
obj.87.body,consequatur omnis est praesentiumducimus non isteneque hic deseruntvoluptatibus veniam cum et rerum sed
obj.88.userId,9
obj.88.id,89
obj.88.title,sint soluta et vel magnam aut ut sed qui
obj.88.body,repellat aut aperiam totam temporibus autem etarchitecto magnam utconsequatur qui cupiditate rerum quia soluta dignissimos nihil iuretempore quas est
obj.89.userId,9
obj.89.id,90
obj.89.title,ad iusto omnis odit dolor voluptatibus
obj.89.body,minus omnis soluta quiaqui sed adipisci voluptates illum ipsam voluptatemeligendi officia ut ineos soluta similique molestias praesentium blanditiis
obj.90.userId,10
obj.90.id,91
obj.90.title,aut amet sed
obj.90.body,libero voluptate eveniet aperiam sedsunt placeat suscipit molestiassimilique fugit nam natusexpedita consequatur consequatur dolores quia eos et placeat
obj.91.userId,10
obj.91.id,92
obj.91.title,ratione ex tenetur perferendis
obj.91.body,aut et excepturi dicta laudantium sint rerum nihillaudantium et ata neque minima officia et similique libero etcommodi voluptate qui
obj.92.userId,10
obj.92.id,93
obj.92.title,beatae soluta recusandae
obj.92.body,dolorem quibusdam ducimus consequuntur dicta aut quo laboriosamvoluptatem quis enim recusandae ut sed suntnostrum est odit totamsit error sed sunt eveniet provident qui nulla
obj.93.userId,10
obj.93.id,94
obj.93.title,qui qui voluptates illo iste minima
obj.93.body,aspernatur expedita soluta quo ab ut similiqueexpedita dolores ametsed temporibus distinctio magnam saepe delenitiomnis facilis nam ipsum natus sint similique omnis
obj.94.userId,10
obj.94.id,95
obj.94.title,id minus libero illum nam ad officiis
obj.94.body,earum voluptatem facere provident blanditiis velit laboriosampariatur accusamus odio saepecumque dolor qui a dicta ab doloribus consequatur omniscorporis cupiditate eaque assumenda ad nesciunt
obj.95.userId,10
obj.95.id,96
obj.95.title,quaerat velit veniam amet cupiditate aut numquam ut sequi
obj.95.body,in non odio excepturi sint eumlabore voluptates vitae quia qui etinventore itaque rerumveniam non exercitationem delectus aut
obj.96.userId,10
obj.96.id,97
obj.96.title,quas fugiat ut perspiciatis vero provident
obj.96.body,eum non blanditiis soluta porro quibusdam voluptasvel voluptatem qui placeat dolores qui velit autvel inventore aut cumque culpa explicabo aliquid atperspiciatis est et voluptatem dignissimos dolor itaque sit nam
obj.97.userId,10
obj.97.id,98
obj.97.title,laboriosam dolor voluptates
obj.97.body,doloremque ex facilis sit sint culpasoluta assumenda eligendi non ut eiussequi ducimus vel quasiveritatis est dolores
obj.98.userId,10
obj.98.id,99
obj.98.title,temporibus sit alias delectus eligendi possimus magni
obj.98.body,quo deleniti praesentium dicta non quodaut est molestiasmolestias et officia quis nihilitaque dolorem quia
obj.99.userId,10
obj.99.id,100
obj.99.title,at nam consequatur ea labore ea harum
obj.99.body,cupiditate quo est a modi nesciunt solutaipsa voluptas error itaque dicta inautem qui minus magnam et distinctio eumaccusamus ratione error aut
"
`;

exports[`Testing Api Testing linter csv 1`] = `
Object {
  "data": Array [
    Array [
      "file.csv",
    ],
  ],
  "errors": Array [
    Object {
      "code": "UndetectableDelimiter",
      "message": "Unable to auto-detect delimiting character; defaulted to ','",
      "row": undefined,
      "type": "Delimiter",
    },
  ],
  "meta": Object {
    "aborted": false,
    "cursor": 8,
    "delimiter": ",",
    "linebreak": "
",
    "truncated": false,
  },
}
`;
